### PR TITLE
Upgrade to Infinispan 13.0 and Cassandra Java Driver 4.14

### PR DESCRIPTION
**Pull request by dominikschlosser**
@tristantarrant We would like to use this store in a project, so i updated the code to the current stable infinispan and cassandra-driver version.

### CODE REVIEW

File: src/main/java/org/infinispan/persistence/cassandra/CassandraStore.java [hunk 1]
```diff
@@ -1,362 +1,429 @@
...
+import java.util.concurrent.Executor;
+import java.util.function.Predicate;
+
+/**
+ * A persistent <code>CacheStore</code> based on Apache Cassandra project. See http://cassandra.apache.org/
+ *
+ * @author Tristan Tarrant
+ * @author Jakub Markos
+ */
+@Store(shared = true)
```
Comment (jabolina) [inline src/main/java/org/infinispan/persistence/cassandra/CassandraStore.java:45]: Add a `@MetaInfServices` here to add this to the services folder.

File: pom.xml [hunk 2]
```diff
@@ -1,302 +1,210 @@
...
+      <maven.compiler.source>11</maven.compiler.source>
+      <maven.compiler.target>11</maven.compiler.target>
+   </properties>
+
+   <dependencies>
+      <dependency>
+         <groupId>org.infinispan</groupId>
+         <artifactId>infinispan-core</artifactId>
+         <scope>provided</scope>
+      </dependency>
```
Comment (jabolina) [inline pom.xml:91]: Pin the version of the dependencies. Some versions can be pulled from the parent pom, e.g. mockito. Checking the effective pom can help.

**Review by jabolina — CHANGES_REQUESTED** [review]
Really nice work! There are a few comments, basically small stuff. 

An additional stylistic one that applies to multiple files, but I'll not hold integration for it, is to revert the changes affecting tab width and the line separator. Just to help with the diff, I see that some files only change the line separator, and others have the diff increased because of tab width.

File: src/main/java/org/infinispan/persistence/cassandra/CassandraStore.java [hunk 3]
```diff
@@ -1,362 +1,429 @@
...
+                    .withDuration(DefaultDriverOption.HEARTBEAT_TIMEOUT, Duration.ofMillis(poolConfig.heartbeatTimeoutMs()))
+                    .withInt(DefaultDriverOption.CONNECTION_POOL_LOCAL_SIZE, poolConfig.localSize())
+                    .withInt(DefaultDriverOption.CONNECTION_POOL_REMOTE_SIZE, poolConfig.remoteSize())
+                    .build();
+
+            CqlSessionBuilder builder = CqlSession.builder();
+            builder.withConfigLoader(configLoader);
+
+            if (configuration.useSsl()) {
+                builder.withSslContext(SSLContext.getDefault());
```
[the remark this answers is not in the document]
Comment (dominikschlosser) [reply src/main/java/org/infinispan/persistence/cassandra/CassandraStore.java:98]: Issue added: #32

File: pom.xml (hunk 2 above)
Comment (dominikschlosser) [reply pom.xml:91]: Thats why i use the infinispan parent pom. it pins all the infinispan-dependencies and thus ensures compatibility. i added explicit versions only for new dependencies, like the cassandra driver or testcontainers-cassandra.

**Review by jabolina — APPROVED** [review]
LGTM!

**Pull request merged**